integration-tests/src/test_loop/utils/transactions.rs

```diff
@@ -39,45 +42,52 @@ pub(crate) fn execute_money_transfers(
         .map(|account| (account.clone(), clients.query_balance(&account)))
         .collect::<HashMap<_, _>>();
     let num_clients = clients.len();
-
-    // Transactions have to be built on top of some block in chain. To make
-    // sure all clients accept them, we select the head of the client with
-    // the smallest height.
-    let (_, anchor_hash) = clients
-        .iter()
-        .map(|client| {
-            let head = client.chain.head().unwrap();
-            (head.height, head.last_block_hash)
-        })
-        .min_by_key(|&(height, _)| height)
-        .unwrap();
     drop(clients);
 
+    let node_data = Arc::new(node_data.to_vec());
+
     for i in 0..accounts.len() {
         let amount = ONE_NEAR * (i as u128 + 1);
-        let sender = &accounts[i];
-        let receiver = &accounts[(i + 1) % accounts.len()];
-        let tx = SignedTransaction::send_money(
-            // TODO: set correct nonce.
-            1,
-            sender.clone(),
-            receiver.clone(),
-            &create_user_test_signer(sender).into(),
-            amount,
-            anchor_hash,
+        let sender = accounts[i].clone();
+        let receiver = accounts[(i + 1) % accounts.len()].clone();
+        let node_data = node_data.clone();
+        *balances.get_mut(&sender).unwrap() -= amount;
+        *balances.get_mut(&receiver).unwrap() += amount;
+        test_loop.send_adhoc_event_with_delay(
```

Comment:
This change is because the previous logic was getting the block hash at the very beginning and using it for all transactions being sent in the future. Now, we query the block hash when sending the transaction instead. This helps the transactions stay within a reasonable transaction_validity_period.